### PR TITLE
corrections and improvements for og:image and og:description

### PR DESCRIPTION
**Pull request by rosskarchner**
- og:image must be a complete URL and can not be an SVG. These are both resolved
- if set in wagtail, og:image now points to the "preview" image
- per discussion on GHE 648, og:description now fails over to the "page title" (under configuration) if "search description" is not set

## Changes

described above

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by sebworks** [comment]
I'm getting `/f/images/homebuyers-blog-5-photo-header.original.png` for the preview image. Should the url be absolute?

**Comment by rosskarchner** [comment]
@sebworks I think that's a reflection of running on localhost, and that it'll be translated into an S3 URL on production (the way the same image behaves when embedded in a post)

@chosak -- I think you worked on that functionality. In production will the 'url' attribute of image objects point to S3?

File: cfgov/jinja2/v1/_layouts/base.html [hunk 1]
```diff
@@ -64,10 +64,17 @@
...
-              content="{{ static('nemo/_/img/logo.svg') }}">
+              content="{{ preview_image.url  }}">
+        {% else %}
+        <meta property="og:image"
+              content="{{request.scheme}}://{{request.get_host()}}{{ static('img/logo.png') }}">
+        {% endif %}
         <!--   Optional -->
-        <meta property="og:description" content="{% block og_desc %}{{- page.search_description if page and page.search_description -}}{% endblock %}">
+        <meta property="og:description" content="{%- block og_desc %}
+        {%- if page %}{{- page.search_description or page.seo_title -}}{% endif %}{% endblock %}">
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_layouts/base.html]: We still need one more fallback here. For example, the Blog index still has a blank `og:description` with this code.

File: cfgov/jinja2/v1/_layouts/base.html [hunk 2]
```diff
@@ -64,10 +64,17 @@
...
         <meta property="og:type" content="{% block og_type %}website{% endblock %}">
         <meta property="og:url" content="{{ request.url }}">
+        {% if page and page.preview_image %}
+        {% set preview_image = image(page.preview_image, 'original') %}
         <meta property="og:image"
-              content="{{ static('nemo/_/img/logo.svg') }}">
+              content="{{ preview_image.url  }}">
+        {% else %}
+        <meta property="og:image"
+              content="{{request.scheme}}://{{request.get_host()}}{{ static('img/logo.png') }}">
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_layouts/base.html]: I'm going to provide you with a new image for this that includes some whitespace around the logo.

File: cfgov/jinja2/v1/_layouts/base.html (hunk 1 above)
Comment (rosskarchner) [inline cfgov/jinja2/v1/_layouts/base.html]: hmm, maybe I'll add page.title

File: cfgov/jinja2/v1/_layouts/base.html (hunk 2 above)
Comment (rosskarchner) [inline cfgov/jinja2/v1/_layouts/base.html]: Cool. Feel free to just commit to the branch if you'd like.

**Comment by chosak** [comment]
@rosskarchner @sebworks sorry, late to this but

> In production will the 'url' attribute of image objects point to S3?

Yes, there's even [a unit test](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/tests/test_s3utils.py#L41) that tests this.

**Comment by Scotchester** [comment]
Confirming that the S3 URL for the preview image works as expected:

screen shot 2017-03-08 at 08 58 23

**Comment by rosskarchner** [comment]
@Scotchester phew!

**Pull request merged**